internal/config.go

```diff
@@ -91,57 +99,56 @@ func (p *ProxyMode) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// ConfigureProxy preparing the server with the proxy configuration that the user has indicated
+func (cfg *Config) ConfigureProxy(proxyMode ProxyMode, proxyURL string) {
+	cfg.Proxy.Mode = proxyMode
+	cfg.Proxy.Url = proxyURL
+}
+
 // ConfigOpt function to encapsulate optional parameters
 type ConfigOpt func(cfg *Config) error
 
 // NewConfig initialize the config
-func NewConfig(impostersPath, host string, port int, opts ...ConfigOpt) (Config, error) {
+func NewConfig(impostersPath, host string, port int) (Config, error) {
+	if impostersPath == "" {
+		return Config{}, errEmptyImpostersPath
+	}
+
+	if host == "" {
+		return Config{}, errEmptyHost
+	}
+
+	if port <= 0 {
```

Comment:
```suggestion
	if port < 0 || port > 65535 {
```

Reply:
According to [RFC 793](https://tools.ietf.org/html/rfc793), the port is a 16-bit unsigned int.

This means the range is 0 - 65535.

However, in the near future, we could potentially add some additional checks (reserved ports, ephemeral ports, etc. However, the range check is the least restrictive check, I guess.

Reply:
or using directly a uint ^^'